### ISSUE TITLE
`softmin` and `softmax` operators support different input dtypes based on whether the `dtype` kwarg is passed

### ISSUE THREAD

**Issue by saketh-are** [pytorch/pytorch#68752]
See OpInfos for `softmax`:

https://github.com/pytorch/pytorch/blob/a2e35e167bc80b7f87ed9e1283e273fc0d89d1d3/torch/testing/_internal/common_methods_invocations.py#L9673

**Comment by jbschlosser** [comment]
Triage review to decide the appropriate labels.

**Comment by ngimel** [comment]
The correct behavior is to support only dtypes that are supported when `dtype` kwarg is not passed. Passing `dtype` kwarg artificially extends supported dtypes because input of any type is then cast to the passed `dtype`.

**Comment by malfet** [comment]
@ngimel we should disallow softmin/max on integral and boolean types even if dtype is specified
